### PR TITLE
Dev new route page and controller

### PR DESCRIPTION
**Pull request by timbez**

### CODE REVIEW

File: project/controller/newRouteController.php [hunk 1]
```diff
@@ -1,58 +1,73 @@
...
 
 $userId = getActiveUserID();
 
-if($userId === null) {
-    //TODO: Session is not initialized, move to login?
-}
-
-if($mode === null) {
-    //TODO: Is this check even needed? Can user somehow hack POST so he does not post one value?
+if (!nullCheck(array($distance, $name, $mode, $team, $startLatitude, $startLongitude, $endLatitude, $endLongitude))) {
```
Comment (ThoraX-SVK) [inline project/controller/newRouteController.php]: Team can be null, when saving public route. Or private route

File: project/controller/newRouteController.php [hunk 2]
```diff
@@ -1,58 +1,73 @@
...
     default:
         break;
 }
 
+redirectToHomePageWithMessage(ROUTE_SUCCESSFULLY_SAVED);
+
+
 function getDataFromPOST($key) {
-    if (isset($_POST[$key])) {
+    if (isset($_POST[$key]) and $_POST[$key] !== "") {
```
Comment (ThoraX-SVK) [inline project/controller/newRouteController.php]: Time does not come as empty string I think. But as 00:00:00. Can we investigate this more?

File: project/controller/newRouteController.php (hunk 1 above)
Comment (timbez) [reply project/controller/newRouteController.php]: oh youre right, my bad

File: project/controller/newRouteController.php (hunk 2 above)
Comment (timbez) [reply project/controller/newRouteController.php]: i will look into this

Comment (ThoraX-SVK) [reply project/controller/newRouteController.php]: Something like php function empty() maybe?

Comment (timbez) [reply project/controller/newRouteController.php]: checked it out, both time and date inputs give empty strings in POST when sent empty

Comment (ThoraX-SVK) [reply project/controller/newRouteController.php]: Any news regarding this? @timbez

**Pull request merged**